### ISSUE TITLE
Want the tf.signal.stft function to return the complete output (i.e. both the positive and negative frequencies)

### ISSUE THREAD

**Issue by karanpareek96** [tensorflow/tensorflow#28244]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 1.12.0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**

The STFT Tensorflow function takes into consideration the Hermitian symmetry of the generated output i.e. it only considers the positive frequency elements. Is there a way to recover the entire pre-processed tensor that contains all the calculated elements?

In other words, is there a way to calculate the STFT of an input tensor and generate a spectrogram that has dimensions `fft_length * frame_length`?

**Will this change the current api? How?**

This will just add an additional feature to the existing function.

**Who will benefit with this feature?**

Anyone who is doing audio processing using Tensorflow.

**Any Other info.**

**Comment by rryan** [comment]
Hi @karanpareek96,

We follow `scipy.signal.stft`'s behavior here, which is to not return the Hermitian symmetric components. This allows us to use an RFFT instead of an FFT, which is generally 2x faster and uses less memory. The Hermitian symmetric component are duplicated, so you could mirror conjugate and reverse the result from `tf.signal.stft`, and concatenate the two together. 

Also, it should be easy to define your own stft that uses FFT instead of RFFT. The implementation of `tf.signal.stft` is quite simple, since it calls other functions: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/signal/spectral_ops.py#L68-L92

Example (that I haven't run, so it might not work :), but hopefully you get the idea).

```python
def complex_input_stft(x, frame_length, frame_step, fft_length, window_fn=tf.signal.hann_window, pad_end):
  frames = tf.signal.frame(x, frame_length, frame_step, pad_end=pad_end)
  frames *= window_fn(frame_length, dtype=frames.dtype)
  return tf.signal.fft(frames, [fft_length])
```
